### PR TITLE
feat: support new `perspective` option for clients

### PR DESCRIPTION
**Pull request by Nelwhix**
I added the new [perspective](https://www.sanity.io/docs/perspectives#70340f55f01d) option to the sanity client config.

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -68,6 +68,10 @@ export default defineNuxtModule<SanityModuleOptions>({
       options.useCdn = process.env.NODE_ENV === 'production' && !options.token
     }
 
+    if (options.perspective === 'previewDrafts' && !options.useCdn) {
+        logger.info("To use perspective: previewDrafts, useCdn must be set to true")
```
Comment (danielroe) [inline src/module.ts:72]: Could we just set the value and print a message letting the user know?

File: src/runtime/client.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ export interface SanityConfiguration {
   apiVersion: string
   withCredentials?: boolean
   token?: string
+  perspective?: 'raw'|'published'|'previewDrafts'
```
Comment (danielroe) [inline src/runtime/client.ts:19]:
```suggestion
  perspective?: 'raw' | 'published' | 'previewDrafts'
```

File: src/module.ts (hunk 1 above)
Comment (danielroe) [inline src/module.ts:72]:
```suggestion
      logger.info("To use perspective: previewDrafts, useCdn must be set to true")
```

Comment (danielroe) [reply src/module.ts]: don't you think there might also be a perspective set per client, e.g. if the user has a custom preview client as well as a main one?

Comment (Nelwhix) [reply src/module.ts]: Yes I see that now, check my implementation in client.ts

Comment (Nelwhix) [reply src/module.ts]: Done

File: src/runtime/client.ts [hunk 3]
```diff
@@ -34,13 +35,19 @@ export const getByteSize = (query: string) =>
...
     dataset,
     apiVersion = '1',
     withCredentials,
     token,
+    perspective = 'raw',
   } = config
+
+  let useCdn = config.useCdn
+  if (perspective === 'previewDrafts' && !useCdn) {
+    useCdn = true
```
Comment (danielroe) [inline src/runtime/client.ts:48]:
```suggestion
  const useCdn = perspective === 'previewDrafts' || config.useCdn
```

File: src/runtime/client.ts [hunk 4]
```diff
@@ -34,13 +35,17 @@ export const getByteSize = (query: string) =>
...
     projectId,
     dataset,
     apiVersion = '1',
     withCredentials,
     token,
+    perspective = 'raw',
   } = config
+
+  const useCdn = perspective === 'previewDrafts' || config.useCdn
+  }
```
Comment (danielroe) [inline src/runtime/client.ts:47]: oops, sorry
```suggestion
  }
```

File: src/module.ts (hunk 1 above)
Comment (danielroe) [reply src/module.ts]: thank you ❤️

nice

**Comment by danielroe** [comment]
Do we also need a new type for module options?

**Comment by Nelwhix** [comment]
I am not sure

**Review by danielroe — APPROVED** [review]
Thank you! ❤️

**Pull request merged**